mono/metadata/image.c

```diff
@@ -1077,6 +1077,102 @@ install_pe_loader (void)
 	mono_install_image_loader (&pe_loader);
 }
 
+/*
+Ignored assemblies.
+
+There are some assemblies we need to ignore because they include an implementation that doesn't work under mono.
+Mono provides its own implementation of those assemblies so it's safe to do so.
+
+The ignored_assemblies list is generated using tools/nuget-hash-extractor and feeding the problematic nugets to it.
+
+Right now the list of nugets are the ones that provide the assemblies in $ignored_assemblies_names.
+
+This is to be removed once a proper fix is shipped through nuget.
+
+*/
+
+typedef enum {
+	SYS_RT_INTEROP_RUNTIME_INFO = 0, //System.Runtime.InteropServices.RuntimeInformation
+	SYS_GLOBALIZATION_EXT = 1, //System.Globalization.Extensions
+	SYS_IO_COMPRESSION = 2, //System.IO.Compression
+	SYS_NET_HTTP = 3, //System.Net.Http
+	SYS_TEXT_ENC_CODEPAGES = 4, //System.Text.Encoding.CodePages
+} IgnoredAssemblyNames;
+
+typedef struct {
+	int hash;
+	int assembly_name;
+	const char guid [40];
+} IgnoredAssembly;
+
+const char *ignored_assemblies_names[] = {
+	"System.Runtime.InteropServices.RuntimeInformation.dll",
+	"System.Globalization.Extensions.dll",
+	"System.IO.Compression.dll",
+	"System.Net.Http.dll",
+	"System.Text.Encoding.CodePages.dll"
+};
+
+#define IGNORED_ASSEMBLY(HASH, NAME, GUID, VER_STR)	{ .hash = HASH, .assembly_name = NAME, .guid = GUID }
+
+static const IgnoredAssembly ignored_assemblies [] = {
+	IGNORED_ASSEMBLY (0x1136045D, SYS_GLOBALIZATION_EXT, "475DBF02-9F68-44F1-8FB5-C9F69F1BD2B1", "4.0.0 net46"),
```

Comment:
@jaredpar yes, it's the MVID. This is a very targeted hack and we do understand it doesn't catch all cases (e.g. preview packages from myget etc).

We're still waiting for a proper solution in corefx: https://github.com/dotnet/corefx/issues/15112